app/models/viewModels/PaymentsHistoryModel.scala

```diff
@@ -30,50 +31,60 @@ case class PaymentsHistoryModel(chargeType: ChargeType,
 
 object PaymentsHistoryModel {
 
-  implicit class JsonOps(json: JsValue) {
-    def get[T](key: String)(implicit reads: Reads[T]): T = json.\(key).validate[T].fold(
-      _ => throw new IllegalStateException(s"The data for key $key could not be found in the Json: ${json.toString}"),
-      identity
-    )
-  }
-
   implicit val writes: Writes[PaymentsHistoryModel] = Json.writes[PaymentsHistoryModel]
 
   implicit val reads: Reads[Seq[PaymentsHistoryModel]] = new Reads[Seq[PaymentsHistoryModel]] {
+
     override def reads(json: JsValue): JsResult[Seq[PaymentsHistoryModel]] = {
 
-    val transactionsList: List[JsValue] = json.get[List[JsValue]](FinancialTransactionsConstants.financialTransactions).filter { transaction =>
-      val transactions = transaction.get[String]("chargeType")
-      ChargeType.isValidChargeType(transactions)
-    }
+      val transactions: Seq[JsValue] = (json \ FinancialTransactionsConstants.financialTransactions).as[Seq[JsValue]]
+      val filteredTransactions: Seq[JsValue] = transactions.filter { transaction =>
+        ChargeType.isValidChargeType((transaction \ FinancialTransactionsConstants.chargeType).as[String])
+      }
+
+      JsSuccess(
+        filteredTransactions flatMap { transaction =>
+
+          val chargeType: ChargeType =  ChargeType.apply((transaction \ FinancialTransactionsConstants.chargeType).as[String])
 
-    def getItemsForPeriod(transaction: JsValue): List[(BigDecimal, Option[LocalDate])] = {
-      transaction.\("items").validate[List[JsValue]].fold(
-        _ => throw new IllegalStateException("The data for key items could not be found in the Json"),
-        list => if (list.nonEmpty) {
-          list.map(item => item.get[BigDecimal]("amount") ->
-            getOptionDate(item, FinancialTransactionsConstants.clearingDate))
-        } else {
-          throw new IllegalStateException("The items list was found but the list was empty")
+          val transactions: Seq[Option[PaymentsHistoryModel]] = getSubItemsForTransaction(transaction) map {
+            subItem => generatePaymentModel(chargeType, subItem, transaction)
+          }
+
+          transactions.flatten
```

Comment:
Could we filter out the transactions without a clearingDate please: `transactions.flatten.filter(_.clearingDate.isDefined)`